### ISSUE TITLE
How can I change vocab size for pretrained model?

### ISSUE THREAD

**Issue by hahmyg** [google-research/bert#406]
Is there way to change (expand) vocab size for pretrained model?

**Comment by maggieezzat** [comment]
Hello, did you find an answer to that?

**Comment by jeandalf** [comment]
if your goal is to regenerate the vocab file, than I believe https://github.com/google/sentencepiece is what you are looking for.

**Comment by maggieezzat** [comment]
If I generate a vocab file as the same size as their vocab file, and then I use it with my own data, can I train from their checkpoint, or I have to train from the scratch?

**Comment by jageshmaharjan** [comment]
@maggieezzat , if you change the vocab.txt file (increase/reduce the tokens), you can't use the same pre-trained model. Because the when running `run_pretraining.py` with different vocab size but the provided pre-trained model, the size of embedding tensor differ to that from the checkpoint.

But if u managed to hack this issue, let me know :)

**Comment by jageshmaharjan** [comment]
I using the pre-trained model, the vocab size is 30522.
We can replace the [unused1] to [unused993] or less with your vocabulary word, make sure in between there are these tokens [UNK] [CLS] [SEP][MASK], which shouldn't be replaced.
The resulting vocab size should be 30522.

**Comment by maggieezzat** [comment]
Hello @jageshmaharjan If you will replace the vocab file, even with the same size, you have to train from scratch, this is because each token in the vocab file, corresponds to a vector (word embedding) learned from the training phase. Now if you change the vocab, different words map to wrong embeddings, that is the issue.

**Comment by maggieezzat** [comment]
Sorry I didnt see your last comment looks like we reached the samr coclusion =)

**Comment by divyashreepathihalli** [comment]
you can now do this. Keras nightly version has added a new util `keras.utils.warmstart_embedding_matrix`. Using this you can continuously train your model with changing vocabulary.  https://www.tensorflow.org/api_docs/python/tf/keras/utils/warmstart_embedding_matrix
